pkg/preparer/orchestrate.go

```diff
@@ -79,6 +79,90 @@ func (p podWorkerID) String() string {
 	return fmt.Sprintf("%s-%s", p.podID.String(), p.podUniqueKey)
 }
 
+func (p *Preparer) InstallWhiteListPods(whiteListPods map[types.PodID]bool, preparerConfig *PreparerConfig) error {
+	intentResults, _, err := p.store.ListPods(consul.INTENT_TREE, p.node)
+	realityResults, _, err := p.store.ListPods(consul.REALITY_TREE, p.node)
+	var intents []consul.ManifestResult
+	if err != nil {
+		p.Logger.WithError(err).Errorln("could not check intent")
+		return err
+	}
+	errorChan := make(chan error)
+	quit := make(chan struct{})
+
+	for _, intentResult := range intentResults {
+		intentSHA, _ := intentResult.Manifest.SHA()
+		if _, ok := whiteListPods[intentResult.Manifest.ID()]; ok {
+			// compare intent/reality for the pod here, to prevent reinstall
+			installed := false
+			for _, realityResult := range realityResults {
+				realitySHA, _ := realityResult.Manifest.SHA()
+				if intentSHA == realitySHA {
+					installed = true
+					break
+				}
+			}
+			if !installed {
+				intents = append(intents, intentResult)
+			}
+		}
+	}
+
+	for _, intentManifest := range intents {
+		intentManifest := &ManifestPair{
```

Comment:
I defer to your judgment, but I wanted point out that I found it tough to read these two lines. You declare a variable `intentManifest` on line 111, type `ManifestResult`, and then reuse the same variable name on the very next line, but now its type is `&ManifestPair`. I found it harder to reason about than I think it has to be.